### ISSUE TITLE
Add ability to set Catalog Number formatter for collections.

### ISSUE THREAD

**Issue by benanhalt** [specify/specify7#2355]
The Catalog Number formatter is available in the schema config, but that configuration is actually not used. Instead it has to be set in the CatalogNumFormat field of the collection table. Currently there is no way of changing it in Specify 6 or 7 short of an SQL update statement.

**Comment by maxxxxxdlp** [comment]
Can't we just add it to the default `Collection` form as a pick list?

The security panel already has a link to the `Collection` record for the current collection

**Comment by benanhalt** [comment]
Yes. The picklist would need to be automatically populated with the formatters from the UIFormatters appresource that are defined for the catalog number field.

**Comment by maxxxxxdlp** [comment]
When implementing this, take at how other custom pick lists have been implemented (like the one for selecting tree ranks on a tree table, or the ones for selecting table/field/resource formatter on a pick list form)

**Comment by grantfitzsimmons** [comment]
I would recommend implementing this so when you select a field formatter in the schema config it automatically updates the collection table with the correct field formatter name.

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/to-change-a-catalog-number-format-other-than-default/283/6
